packages/api-client/src/notification/NotificationAPI/NotificationsAPI.test.ts

```diff
@@ -75,5 +78,28 @@ describe('NotificationAPI', () => {
       expect(result.notifications.length).toBe(mockedResultData.notifications.length);
       expect(result.missedNotification).toBe(mockedNotificationId);
     });
+
+    it('should return all the notifications from the beginning for status code 404', async () => {
+      const ErrorResponse = new BackendError('Some notifications not found', BackendErrorLabel.NOT_FOUND, 404);
+
+      //first call returns not found backend error
+      //function gets called again without 'since' param under the hood
+      //second call returns all the notifications with id of missed notification
+      jest
+        .spyOn(client, 'sendJSON')
+        .mockImplementationOnce(() => Promise.reject(ErrorResponse))
+        .mockImplementationOnce(() =>
+          Promise.resolve<AxiosResponse>({
+            status: 200,
+            data: {...mockedResultData},
+          } as AxiosResponse),
+        );
```

Comment:
`getNotificationChunks` recursive function that makes the actual api call is called under the hood inside `getAllNotifications`. When backend not found error is caught, we manually send another call to the same endpoint without `since` param, what should give use full list of notifications in the response.